### ISSUE TITLE
Crash on AMD ROCm: gfx1151 is listed as supported on Linux, but its data files are missing

### ISSUE THREAD

**Issue by ArcticLight** [lmstudio-ai/lmstudio-bug-tracker#1269]
**Which version of LM Studio?**
LM Studio 0.3.33

**Which operating system?**

Linux (Ubuntu on the 6.14.0-OEM kernel, as per AMD's install instructions for the platform Halo Strix)

- Note that ROCm and ROCBlas work properly inside Docker after following AMD's install instructions 

**What is the bug?**
AMD ROCm backend declares support for platform `gfx1151` in the manifest, but the data files for it are not included in the vendor directory, causing the backend to crash on inference.

**Screenshots**

<img width="433" height="341" alt="Image" src="https://github.com/user-attachments/assets/59a021d9-567c-40c5-8659-1f35d957ea52" />
<img width="993" height="68" alt="Image" src="https://github.com/user-attachments/assets/5351362d-25cd-44f5-9c90-c8f468e5b172" />
<img width="1041" height="577" alt="Image" src="https://github.com/user-attachments/assets/04e58f9e-4d6c-4aa2-b03a-6100069c61ca" />

**Logs**

```
rocBLAS error: Cannot read /home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary.dat: Illegal seek for GPU arch : gfx1151
2025-12-02 17:45:05 [DEBUG]
  List of available TensileLibrary Files :
2025-12-02 17:45:05 [DEBUG]
 "/home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx942.dat"
2025-12-02 17:45:05 [DEBUG]
 "/home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1101.dat"
2025-12-02 17:45:05 [DEBUG]
 "/home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1201.dat"
2025-12-02 17:45:05 [DEBUG]
 "/home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1102.dat"
2025-12-02 17:45:05 [DEBUG]
 "/home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx908.dat"
2025-12-02 17:45:05 [DEBUG]
 "/home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1200.dat"
2025-12-02 17:45:05 [DEBUG]
 "/home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx90a.dat"
2025-12-02 17:45:05 [DEBUG]
 "/home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1100.dat"
"/home/xyzzy/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1030.dat"
```

**To Reproduce**
Run any model on a Strix Halo system

**Comment by ubiclouder** [comment]
Same problem :
37:15 [DEBUG]
 Sampling params:	repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
	top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.800
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
Sampling: 
logits -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
Generate: n_ctx = 8960, n_batch = 512, n_predict = -1, n_keep = 89
Total prompt tokens: 89
Prompt tokens to decode: 89
BeginProcessingPrompt
2026-01-06 14:37:15 [DEBUG]
 rocBLAS error: Cannot read /home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary.dat: Illegal seek for GPU arch : gfx1151
 List of available TensileLibrary Files : 
"/home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1100.dat"
2026-01-06 14:37:15 [DEBUG]
 "/home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1101.dat"
"/home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx908.dat"
2026-01-06 14:37:15 [DEBUG]
 "/home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx90a.dat"
2026-01-06 14:37:15 [DEBUG]
 "/home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1030.dat"
2026-01-06 14:37:15 [DEBUG]
 "/home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1200.dat"
2026-01-06 14:37:15 [DEBUG]
 "/home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1102.dat"
"/home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx942.dat"
2026-01-06 14:37:15 [DEBUG]
 "/home/ubiclouder/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1201.dat"

**Comment by snippetsBySam** [comment]
This is still an issue:

```
[2026-01-08 20:56:32][DEBUG] Sampling params:	repeat_last_n = 64, repeat_penalty = 1.050, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
	top_k = 20, top_p = 0.800, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.700
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
[2026-01-08 20:56:32][DEBUG] Sampling: 
logits -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
Generate: n_ctx = 64000, n_batch = 512, n_predict = -1, n_keep = 9
[2026-01-08 20:56:32][DEBUG] Total prompt tokens: 9
Prompt tokens to decode: 9
BeginProcessingPrompt
[2026-01-08 20:56:32][DEBUG] 
rocBLAS error: Cannot read /home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary.dat: Illegal seek for GPU arch : gfx1151
 List of available TensileLibrary Files :
[2026-01-08 20:56:32][DEBUG] "/home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1100.dat"
[2026-01-08 20:56:32][DEBUG] "/home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1102.dat"
"/home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx90a.dat"
"/home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1200.dat"
[2026-01-08 20:56:32][DEBUG] "/home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx908.dat"
[2026-01-08 20:56:32][DEBUG] "/home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx942.dat"
[2026-01-08 20:56:32][DEBUG] "/home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1101.dat"
"/home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1201.dat"
[2026-01-08 20:56:32][DEBUG] "/home/sam/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/TensileLibrary_lazy_gfx1030.dat"
```

**Comment by snippetsBySam** [comment]
It appears all that is needed to fix this is to include the tenstile files for gfx1151. The folder contains files for other GPU architectures, just not the gfx1151 one

<img width="394" height="319" alt="Image" src="https://github.com/user-attachments/assets/f2d2d813-c6d1-47a1-a650-69c77cab0f62" />

**Comment by goniz** [comment]
Yes! this is very unfortunate :(
waiting for a fix as well

**Comment by wdyun** [comment]
Please note that copying only TensileLibrary_lazy_gfx1151.dat might not be enough. You should copy all files related to gfx1151 to ensure everything works correctly.

Steps:

Run the following command to copy all necessary files:

Bash

cp /opt/rocm-7.1.1/lib/hipblaslt/library/*gfx1151* ~/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas
Restart your application or system.

Try running the inference again.

This should resolve the issue for gfx1151 users.

**Comment by snippetsBySam** [comment]
> Please note that copying only TensileLibrary_lazy_gfx1151.dat might not be enough. You should copy all files related to gfx1151 to ensure everything works correctly.
> 
> Steps:
> 
> Run the following command to copy all necessary files:
> 
> Bash
> 
> cp /opt/rocm-7.1.1/lib/hipblaslt/library/_gfx1151_ ~/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas Restart your application or system.
> 
> Try running the inference again.
> 
> This should resolve the issue for gfx1151 users.

Does that work for you? I just tried that and it didn't work for me - still crashing. Also your command copies hipblaslt to rocblas
I did `cp -r /opt/rocm-7.1.1/lib/rocblas/library/ ~/.lmstudio/extensions/backends/vendor/linux-llama-rocm-vendor-v3/rocblas/library/`

**Comment by vercingetorx** [comment]
Confirmed getting these errors with Strix Halo. Have to use Vulkan runtime.

**Comment by snippetsBySam** [comment]
I have found the root cause this issue, but it is up to lmstudio to fix. The issue is that the version of ROCm they are bundling with their runtime is 6.4.1 which is very old compared to the latest version of 7.2. Hence the newly added GPUs supported after 6.4.1 will not work with ROCm (including strix halo unfortunately)

<img width="325" height="60" alt="Image" src="https://github.com/user-attachments/assets/ce775d81-7e64-4f22-80b4-44445efe487e" />

**Comment by Hotrhodium** [comment]
Yeah this is kinda unfortunate. Use Ubuntu 25 and mainline the kernel to get latest support, Fight bugs for the strix halo but get everything working (-WiFi lol, causes kernel panic after about 5 hrs, no idea if that will ever get fixed). No ROCm support yet so have to use LM Studio to use Vulcan. Oh hey, ROCm 7 fixes everything! Ok install ROCm carefully because we are way in the weeds now. Oh LM Studio doesn't include gfx-1151.dat files... No problem I brought my own! LM Studio: not so fast, was that for ROCm 6.4? Meanwhile AMD is over here begging me to use ROCm. For anyone else here this may help. Got Qwen 30b working finally on ROCm and lost 25% TPS over Vulcan. Cannot load Qwen Next 80b or OpenAIOSS120b. Use Vulcan and give AMD another 2 years to fix this mess. I love you AMD but this whole roll out was ugly.

**Comment by dracid** [comment]
Honesty, I am considering to return my Framework Desktop over this. What a mess.
Forget that WIFI drops overnight and back USB3 ports stop working after a while... this driver and ROCm situation is just embarrassing.

**Comment by waltercool** [comment]
> Honesty, I am considering to return my Framework Desktop over this. What a mess. Forget that WIFI drops overnight and back USB3 ports stop working after a while... this driver and ROCm situation is just embarrassing.

You have two options:
- Move to the Vulkan backend.
- Use Ollama.

Also, worth to mention this is the support chart from AMD https://rocm.docs.amd.com/en/latest/how-to/system-optimization/strixhalo.html#required-kernel-version
